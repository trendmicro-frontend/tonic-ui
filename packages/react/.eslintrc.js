module.exports = {
  extends: 'trendmicro',
  parser: '@babel/eslint-parser',
  env: {
    browser: true,
    node: true,
    jest: true,
  },
  plugins: [
    '@babel',
    'react-hooks',
  ],
  rules: {
    'react/prop-types': 0,
    'react-hooks/rules-of-hooks': 'error', // Checks rules of Hooks
    'react-hooks/exhaustive-deps': 'error', // Checks effect dependencies
  },
};
