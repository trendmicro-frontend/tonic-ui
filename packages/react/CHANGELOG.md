# @tonic-ui/react

## 2.0.3

### Patch Changes

- fix(react/OverflowTooltip): resolve tooltip misalignment issue with popup menu items by [@tinaClin](https://github.com/tinaClin) in [#894](https://github.com/trendmicro-frontend/tonic-ui/pull/894)

- feat(react): improve tooltip placement and offset calculations by [@cheton](https://github.com/cheton) in [#897](https://github.com/trendmicro-frontend/tonic-ui/pull/897)
