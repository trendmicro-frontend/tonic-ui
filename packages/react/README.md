React UI components built with Emotion and Styled System
