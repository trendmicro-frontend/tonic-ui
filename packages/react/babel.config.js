module.exports = {
  extends: '@trendmicro/babel-config',
  presets: [
    '@babel/preset-env',
    ['@babel/preset-react', {
      /**
       * Decides which runtime to use.
       * - 'automatic' auto imports the functions that JSX transpiles to.
       * - 'classic' does not automatic import anything.
       */
      'runtime': 'classic', // defaults to classic
      //'runtime': 'automatic', // for React 16.14.0, React 17 and higher that supports the new transform
    }],
    '@emotion/babel-preset-css-prop',
  ],
};
