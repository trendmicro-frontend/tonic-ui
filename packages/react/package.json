{
  "name": "@tonic-ui/react",
  "version": "1.0.0-alpha",
  "description": "React Tonic UI component library.",
  "main": "dist/index.js",
  "module": "dist/es/index.js",
  "publishConfig": {
    "access": "public",
    "directory": "build"
  },
  "scripts": {
    "build": "cross-env PACKAGE_NAME=index INPUT=src OUTPUT_DIRECTORY=dist rollup --config rollup.config.js",
    "clean": "del build coverage dist",
    "eslint": "eslint --ext .js,.jsx,.mjs .",
    "postbuild": "cpy --parents {dist/**,LICENSE,README.md,.npmignore,package.json} build",
    "pre-push": "yarn run eslint",
    "prepublish": "yarn run build",
    "test": "jest"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "1.x",
    "@emotion/react": "11.x",
    "@emotion/styled": "11.x",
    "@popperjs/core": "2.x",
    "@tonic-ui/react-hooks": "^1.0.0-alpha",
    "@tonic-ui/styled-system": "^1.0.0-alpha",
    "@tonic-ui/theme": "^1.0.0-alpha",
    "@trendmicro/tmicon": "^1.21.0",
    "chained-function": "^0.5.0",
    "color": "3.x",
    "ensure-type": "1.x",
    "lodash.get": "4.x",
    "micro-memoize": "4.x",
    "prop-types": "15.x",
    "react-focus-lock": "2.x",
    "react-is": "^16.8.0 || ^17",
    "react-transition-group": "4.x"
  },
  "devDependencies": {
    "@babel/cli": "^7.0.0",
    "@babel/core": "^7.0.0",
    "@babel/eslint-parser": "^7.0.0",
    "@babel/eslint-plugin": "^7.0.0",
    "@babel/plugin-transform-runtime": "^7.0.0",
    "@babel/preset-env": "^7.0.0",
    "@babel/preset-react": "^7.0.0",
    "@emotion/babel-preset-css-prop": "~11.2.0",
    "@emotion/jest": "~11.5.0",
    "@rollup/plugin-node-resolve": "~11.2.0",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "@trendmicro/babel-config": "^1.0.2",
    "babel-loader": "~8.2.3",
    "cpy-cli": "~3.1.1",
    "cross-env": "~7.0.3",
    "del-cli": "~4.0.1",
    "eslint": "^7.0.0",
    "eslint-config-trendmicro": "^2.1.0",
    "eslint-loader": "~4.0.2",
    "eslint-plugin-import": "latest",
    "eslint-plugin-jsx-a11y": "latest",
    "eslint-plugin-react": "latest",
    "eslint-plugin-react-hooks": "latest",
    "jest": "~27.3.1",
    "jest-extended": "^1.1.0",
    "react": "^16.8.0 || ^17",
    "react-dom": "^16.8.0 || ^17",
    "react-test-renderer": "^16.8.0 || ^17",
    "rollup": "~2.42.3",
    "rollup-plugin-babel": "~4.4.0"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17",
    "react-dom": "^16.8.0 || ^17"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.js"
    ],
    "coverageReporters": [
      "lcov",
      "text",
      "html"
    ],
    "setupFiles": [],
    "setupFilesAfterEnv": [
      "<rootDir>/jest-setup.js"
    ],
    "testEnvironment": "jsdom",
    "testMatch": [
      "<rootDir>/src/**/*.test.js",
      "<rootDir>/test/**/*.js"
    ]
  },
  "author": "Cheton Wu <cheton_wu@trendmicro.com>",
  "homepage": "https://github.com/trendmicro-frontend/tonic-ui",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/trendmicro-frontend/tonic-ui.git",
    "directory": "packages/react"
  },
  "bugs": {
    "url": "https://github.com/trendmicro-frontend/tonic-ui/issues"
  },
  "keywords": [
    "components",
    "emotion",
    "react",
    "styled-system",
    "tonic-ui"
  ]
}
