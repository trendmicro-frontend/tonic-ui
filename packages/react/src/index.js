// components
export * from './accordion';
export * from './alert';
export * from './badge';
export * from './box';
export * from './button';
export * from './checkbox';
export * from './color-mode';
export * from './color-style';
export * from './css-baseline';
export * from './divider';
export * from './drawer';
export * from './flat-button';
export * from './flex';
export * from './grid';
export * from './icon';
export * from './image';
export * from './input';
export * from './link';
export * from './menu';
export * from './modal';
export * from './pagination';
export * from './popover';
export * from './popper';
export * from './portal';
export * from './presence';
export * from './radio';
export * from './search-input';
export * from './scrollbar';
export * from './select';
export * from './skeleton';
export * from './space';
export * from './spinner';
export * from './stack';
export * from './switch';
export * from './table';
export * from './tabs';
export * from './tag';
export * from './text';
export * from './textarea';
export * from './theme';
export * from './toast';
export * from './tooltip';
export * from './transitions';
export * from './visually-hidden';

// hooks
export useDisclosure from './deprecated/useDisclosure'; // deprecated

// settings
export colorStyle from './shared/color-style';
export theme from './shared/theme';
