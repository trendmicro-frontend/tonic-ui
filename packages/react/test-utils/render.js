import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { toHaveNoViolations } from 'jest-axe';
import * as React from 'react';
import { TonicProvider, theme } from '../src';

expect.extend(toHaveNoViolations);

const customRender = (ui, options) => {
  let wrapper = TonicProvider;
  const useCSSVariables = options?.useCSSVariables;
  if (useCSSVariables) {
    wrapper = ({ children }) => (
      <TonicProvider
        theme={{
          ...theme,
          config: {
            ...theme.config,
            useCSSVariables: true,
          },
        }}
      >
        {children}
      </TonicProvider>
    );
  }

  return render(ui, { wrapper, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
