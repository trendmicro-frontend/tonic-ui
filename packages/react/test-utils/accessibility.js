import { act } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import * as React from 'react';
import { render } from './render';

expect.extend(toHaveNoViolations);

const testA11y = async (ui, options) => {
  const { axeOptions, ...rest } = { ...options };
  const container = React.isValidElement(ui)
    ? render(ui, rest).container
    : ui;
  await act(async () => {
    const results = await axe(container, axeOptions);
    expect(results).toHaveNoViolations();
  });
};

export {
  testA11y,
};
