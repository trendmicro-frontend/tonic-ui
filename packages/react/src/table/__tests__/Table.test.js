import { fireEvent, screen } from '@testing-library/react';
import { render } from '@tonic-ui/react/test-utils/render';
import { testA11y } from '@tonic-ui/react/test-utils/accessibility';
import {
  Table,
  TableHeader,
  TableHeaderRow,
  TableHeaderCell,
  TableBody,
  TableRow,
  TableCell,
  TableColumnResizeHandle,
  TableScrollbar,
  Truncate,
} from '@tonic-ui/react/src';
import React, { useCallback, useMemo, useRef } from 'react';

describe('Table', () => {
  const columns = [
    {
      id: 'priority',
      header: 'Priority',
      accessorKey: 'priority',
      cell: (row) => row.priority ?? '-',
      size: 80,
    },
    {
      id: 'policy',
      header: 'Policy',
      accessorKey: 'policy',
      size: 240,
    },
    {
      id: 'modifiedTime',
      header: 'Last Modified',
      cell: (row) => {
        const mtime = new Date(row.modifiedTime);
        return (
          <Truncate>
            {mtime.toLocaleString()}
          </Truncate>
        );
      },
      size: 180,
    },
    {
      id: 'modifiedBy',
      header: 'Last Editor',
      accessorKey: 'modifiedBy',
      size: 150,
    },
  ];

  const data = [
    { id: 1, priority: 1, policy: 'Team Managers', modifiedTime: 1625875200000, modifiedBy: 'admin' },
    { id: 2, priority: 2, policy: 'Marketing Team', modifiedTime: 1625875200000, modifiedBy: 'admin' },
    { id: 3, priority: 3, policy: 'Sales Department', modifiedTime: 1625875200000, modifiedBy: 'admin' },
    { id: 4, priority: 4, policy: 'Development Team', modifiedTime: 1625875200000, modifiedBy: 'admin' },
    { id: 5, priority: 5, policy: 'IT Department', modifiedTime: 1625875200000, modifiedBy: 'admin' },
    { id: 6, priority: null, policy: 'Server policy (Default)', modifiedTime: 1625097600000, modifiedBy: 'admin' },
    { id: 7, priority: null, policy: 'Endpoint policy (Default)', modifiedTime: 1625097600000, modifiedBy: 'admin' },
  ];

  it('should render correctly with `flexbox` layout', async () => {
    const renderOptions = {
      useCSSVariables: true,
    };

    const { container } = render((
      <Table layout="flexbox">
        <TableHeader>
          <TableHeaderRow>
            <TableHeaderCell>
              Header Cell 1
            </TableHeaderCell>
            <TableHeaderCell>
              Header Cell 2
            </TableHeaderCell>
          </TableHeaderRow>
        </TableHeader>
        <TableBody>
          <TableRow>
            <TableCell>
              Row 1, Cell 1
            </TableCell>
            <TableCell>
              Row 1, Cell 2
            </TableCell>
          </TableRow>
          <TableRow>
            <TableCell>
              Row 2, Cell 1
            </TableCell>
            <TableCell>
              Row 2, Cell 2
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    ), renderOptions);

    expect(container).toMatchSnapshot();

    await testA11y(container);
  });

  it('should render correctly with `table` layout', async () => {
    const renderOptions = {
      useCSSVariables: true,
    };

    const { container } = render((
      <Table layout="table">
        <TableHeader>
          <TableHeaderRow>
            <TableHeaderCell>
              Header Cell 1
            </TableHeaderCell>
            <TableHeaderCell>
              Header Cell 2
            </TableHeaderCell>
          </TableHeaderRow>
        </TableHeader>
        <TableBody>
          <TableRow>
            <TableCell>
              Row 1, Cell 1
            </TableCell>
            <TableCell>
              Row 1, Cell 2
            </TableCell>
          </TableRow>
          <TableRow>
            <TableCell>
              Row 2, Cell 1
            </TableCell>
            <TableCell>
              Row 2, Cell 2
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    ), renderOptions);

    expect(container).toMatchSnapshot();

    await testA11y(container);
  });

  it('should resize columns correctly', async () => {
    const TableComponent = ({
      columns,
      data,
      onColumnResizeStart: onColumnResizeStartProp,
      onColumnResize: onColumnResizeProp,
      onColumnResizeEnd: onColumnResizeEndProp,
    }) => {
      const rows = useMemo(() => {
        return data.map((row, rowIndex) => {
          return {
            id: rowIndex,
            cells: columns.map((column, columnIndex) => {
              return {
                id: `${columnIndex}_${column.id}`,
                column,
                row,
              };
            }),
            row,
          };
        });
      }, [columns, data]);

      const handleResizeStartBy = (columnId) => (coord) => {
        onColumnResizeStartProp(columnId, coord);
      };
      const handleResizeBy = (columnId) => (coord) => {
        onColumnResizeProp(columnId, coord);
      };
      const handleResizeEndBy = (columnId) => (coord) => {
        onColumnResizeEndProp(columnId, coord);
      };

      const layout = 'flexbox'; // One of: 'flexbox', 'table'
      const variant = 'outline'; // One of: 'default', 'outline'

      return (
        <Table
          layout={layout}
          variant={variant}
        >
          <TableHeader>
            <TableHeaderRow>
              {columns.map((column, index) => {
                const styleProps = {
                  flex: 'none',
                  position: 'relative',
                  width: column.size,
                  ...column.style,
                };
                const resizeHandleStyle = {
                  position: 'absolute',
                  top: 0,
                  right: 0,
                  height: '100%',
                };

                return (
                  <TableHeaderCell key={column.id} {...styleProps}>
                    <Truncate>
                      {column.header}
                    </Truncate>
                    <TableColumnResizeHandle
                      data-testid={`column-resize-handle-${index}`}
                      style={resizeHandleStyle}
                      onResizeStart={handleResizeStartBy(column.id)}
                      onResize={handleResizeBy(column.id)}
                      onResizeEnd={handleResizeEndBy(column.id)}
                    />
                  </TableHeaderCell>
                );
              })}
            </TableHeaderRow>
          </TableHeader>
          <TableBody>
            {rows.map(row => (
              <TableRow key={row.id}>
                {row.cells.map((cell, cellIndex) => {
                  const styleProps = {
                    flex: 'none',
                    width: cell.column.size,
                    ...cell.column.style,
                  };
                  const renderCell = (cell) => {
                    if (typeof cell.column.cell === 'function') {
                      return cell.column.cell(row.row);
                    }
                    return (
                      <Truncate>
                        {cell.row[cell.column.accessorKey]}
                      </Truncate>
                    );
                  };

                  return (
                    <TableCell key={cell.id} {...styleProps}>
                      {renderCell(cell)}
                    </TableCell>
                  );
                })}
              </TableRow>
            ))}
          </TableBody>
        </Table>
      );
    };

    const onColumnResizeStart = jest.fn();
    const onColumnResize = jest.fn();
    const onColumnResizeEnd = jest.fn();

    render(
      <TableComponent
        columns={columns}
        data={data}
        onColumnResizeStart={onColumnResizeStart}
        onColumnResize={onColumnResize}
        onColumnResizeEnd={onColumnResizeEnd}
      />
    );

    const el = screen.getByTestId('column-resize-handle-0');

    // Mouse events
    await fireEvent.mouseDown(el, { clientX: 150, clientY: 18 }); // => onResizeStart
    await fireEvent.mouseMove(el, { clientX: 140, clientY: 18 }); // => onResize
    await fireEvent.mouseUp(el, { clientX: 130, clientY: 18 }); // => onResizeEnd
    expect(onColumnResizeStart).toHaveBeenLastCalledWith(
      columns[0].id,
      expect.objectContaining({
        clientX: 150,
        clientY: 18,
      }),
    );
    expect(onColumnResize).toHaveBeenLastCalledWith(
      columns[0].id,
      expect.objectContaining({
        clientX: 140,
        clientY: 18,
      }),
    );
    expect(onColumnResizeEnd).toHaveBeenLastCalledWith(
      columns[0].id,
      expect.objectContaining({
        clientX: 130,
        clientY: 18,
      }),
    );

    // Touch events
    await fireEvent.touchStart(el, { touches: [{ clientX: 120, clientY: 18 }] }); // => onResizeStart
    await fireEvent.touchMove(el, { touches: [{ clientX: 110, clientY: 18 }] }); // => onResize
    await fireEvent.touchEnd(el, { touches: [{ clientX: 100, clientY: 18 }] }); // => onResizeEnd
    expect(onColumnResizeStart).toHaveBeenLastCalledWith(
      columns[0].id,
      expect.objectContaining({
        clientX: 120,
        clientY: 18,
      }),
    );
    expect(onColumnResize).toHaveBeenLastCalledWith(
      columns[0].id,
      expect.objectContaining({
        clientX: 110,
        clientY: 18,
      }),
    );
    expect(onColumnResizeEnd).toHaveBeenLastCalledWith(
      columns[0].id,
      expect.objectContaining({
        clientX: 100,
        clientY: 18,
      }),
    );
  });

  it('should scroll rows correctly', async () => {
    const TableComponent = ({
      columns,
      data,
      onScroll: onScrollProp,
    }) => {
      const rows = useMemo(() => {
        return data.map((row, rowIndex) => {
          return {
            id: rowIndex,
            cells: columns.map((column, columnIndex) => {
              return {
                id: `${columnIndex}_${column.id}`,
                column,
                row,
              };
            }),
            row,
          };
        });
      }, [columns, data]);

      const tableHeaderRef = useRef();

      const onScrollTableBody = useCallback((event) => {
        onScrollProp?.(event);

        const tableHeader = tableHeaderRef?.current;
        if (!tableHeader) {
          return;
        }
        // Sync the scroll position of the table header and table body for horizontal scrolling
        const scrollLeft = event.target.scrollLeft;
        if (tableHeader.scrollleft !== scrollLeft) {
          tableHeader.scrollLeft = scrollLeft;
        }
      }, [onScrollProp]);

      const cellHeight = 36;
      const headerHeight = 36;
      const rowsToDisplay = 5;
      const tableStyleProps = {
        height: headerHeight + rowsToDisplay * cellHeight,
        // Uncomment the following line to specify the width and enable horizontal scrollbar
        width: 200,
      };

      // Sync the scroll position of the table header and table body for horizontal scrolling
      const tableHeaderStyleProps = {
        overflow: 'hidden',
      };

      return (
        <Table
          layout="flexbox"
          variant="default"
          {...tableStyleProps}
        >
          <TableHeader
            ref={tableHeaderRef}
            {...tableHeaderStyleProps}
          >
            <TableHeaderRow>
              {columns.map(column => {
                const styleProps = {
                  flex: 'none',
                  width: column.size,
                  ...column.style,
                };

                return (
                  <TableHeaderCell key={column.id} {...styleProps}>
                    <Truncate>
                      {column.header}
                    </Truncate>
                  </TableHeaderCell>
                );
              })}
            </TableHeaderRow>
          </TableHeader>
          <TableScrollbar
            data-testid="scrollbar"
            height="100%"
            overflow="visible" // Make the scrollbar visible
            onScroll={onScrollTableBody} // To sync the scroll position of the table header and table body for horizontal scrolling
          >
            <TableBody>
              {rows.map(row => (
                <TableRow key={row.id}>
                  {row.cells.map((cell, cellIndex) => {
                    const styleProps = {
                      flex: 'none',
                      width: cell.column.size,
                      ...cell.column.style,
                    };
                    const renderCell = (cell) => {
                      if (typeof cell.column.cell === 'function') {
                        return cell.column.cell(row.row);
                      }
                      return (
                        <Truncate>
                          {cell.row[cell.column.accessorKey]}
                        </Truncate>
                      );
                    };

                    return (
                      <TableCell key={cell.id} {...styleProps}>
                        {renderCell(cell)}
                      </TableCell>
                    );
                  })}
                </TableRow>
              ))}
            </TableBody>
          </TableScrollbar>
        </Table>
      );
    };

    const onScroll = jest.fn();

    render(
      <TableComponent
        columns={columns}
        data={data}
        onScroll={onScroll}
      />
    );

    const scrollContainer = screen.getByTestId('scrollbar').firstChild;
    await fireEvent.scroll(scrollContainer, {
      target: {
        scrollTop: 50,
      },
    });
    await fireEvent.scroll(scrollContainer, {
      target: {
        scrollLeft: 50,
      },
    });
    expect(onScroll).toHaveBeenCalledTimes(2);
  });
});
