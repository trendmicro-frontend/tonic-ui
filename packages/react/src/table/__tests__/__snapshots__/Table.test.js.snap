// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly with \`flexbox\` layout 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.emotion-6 {
  border-bottom: var(--tonic-borders-2);
  border-bottom-color: var(--tonic-colors-gray-30);
  color: var(--tonic-colors-black-secondary);
  font-weight: var(--tonic-fontWeights-semibold);
  padding-left: var(--tonic-space-3x);
  padding-right: var(--tonic-space-3x);
  padding-top: var(--tonic-space-2x);
  padding-bottom: calc(.5rem - .125rem);
  width: 150px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.emotion-14 {
  border-bottom: var(--tonic-borders-1);
  border-bottom-color: var(--tonic-colors-gray-30);
  color: var(--tonic-colors-black-primary);
  padding-left: var(--tonic-space-3x);
  padding-right: var(--tonic-space-3x);
  padding-top: var(--tonic-space-2x);
  padding-bottom: calc(.5rem - .0625rem);
  width: 150px;
}

<div>
  <div
    class="emotion-0 emotion-1"
    role="table"
  >
    <div
      class="emotion-2 emotion-1"
      role="rowgroup"
    >
      <div
        class="emotion-4 emotion-1"
        role="row"
      >
        <div
          class="emotion-6 emotion-1"
          role="columnheader"
        >
          Header Cell 1
        </div>
        <div
          class="emotion-6 emotion-1"
          role="columnheader"
        >
          Header Cell 2
        </div>
      </div>
    </div>
    <div
      class="emotion-10 emotion-1"
      role="rowgroup"
    >
      <div
        class="emotion-12 emotion-1"
        role="row"
      >
        <div
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 1, Cell 1
        </div>
        <div
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 1, Cell 2
        </div>
      </div>
      <div
        class="emotion-12 emotion-1"
        role="row"
      >
        <div
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 2, Cell 1
        </div>
        <div
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 2, Cell 2
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table should render correctly with \`table\` layout 1`] = `
.emotion-0 {
  border-collapse: collapse;
  border-spacing: 0;
  display: table;
}

.emotion-2 {
  display: table-header-group;
}

.emotion-4 {
  display: table-row;
}

.emotion-6 {
  display: table-cell;
  text-align: start;
  border-bottom: var(--tonic-borders-2);
  border-bottom-color: var(--tonic-colors-gray-30);
  color: var(--tonic-colors-black-secondary);
  font-weight: var(--tonic-fontWeights-semibold);
  padding-left: var(--tonic-space-3x);
  padding-right: var(--tonic-space-3x);
  padding-top: var(--tonic-space-2x);
  padding-bottom: calc(.5rem - .125rem);
  width: 150px;
}

.emotion-10 {
  display: table-row-group;
}

.emotion-12 {
  display: table-row;
}

.emotion-14 {
  display: table-cell;
  border-bottom: var(--tonic-borders-1);
  border-bottom-color: var(--tonic-colors-gray-30);
  color: var(--tonic-colors-black-primary);
  padding-left: var(--tonic-space-3x);
  padding-right: var(--tonic-space-3x);
  padding-top: var(--tonic-space-2x);
  padding-bottom: calc(.5rem - .0625rem);
  width: 150px;
}

<div>
  <table
    class="emotion-0 emotion-1"
    role="table"
  >
    <thead
      class="emotion-2 emotion-1"
      role="rowgroup"
    >
      <tr
        class="emotion-4 emotion-1"
        role="row"
      >
        <th
          class="emotion-6 emotion-1"
          role="columnheader"
        >
          Header Cell 1
        </th>
        <th
          class="emotion-6 emotion-1"
          role="columnheader"
        >
          Header Cell 2
        </th>
      </tr>
    </thead>
    <tbody
      class="emotion-10 emotion-1"
      role="rowgroup"
    >
      <tr
        class="emotion-12 emotion-1"
        role="row"
      >
        <td
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 1, Cell 1
        </td>
        <td
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 1, Cell 2
        </td>
      </tr>
      <tr
        class="emotion-12 emotion-1"
        role="row"
      >
        <td
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 2, Cell 1
        </td>
        <td
          class="emotion-14 emotion-1"
          role="cell"
        >
          Row 2, Cell 2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
